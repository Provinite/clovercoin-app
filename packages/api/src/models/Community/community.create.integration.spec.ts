import { testClient } from "../../test/testClient";
import { expect, describe, it } from "@jest/globals";

describe("create commuity", () => {
  it("creates a community", async () => {
    const createdCommunity = await testClient.createCommunity({
      variables: {
        input: {
          name: "Test Community",
        },
      },
    });
    expect(createdCommunity.errors).toBeFalsy();

    await expect(createdCommunity).toMatchSnapshot({
      data: {
        createCommunity: {
          id: expect.any(String),
        },
      },
    });

    const fetchedCommunity = await testClient.getCommunity({
      variables: {
        filters: {
          id: createdCommunity.data!.createCommunity.id,
        },
      },
    });
    expect(fetchedCommunity.errors).toBeFalsy();

    expect(fetchedCommunity.data!.community).toEqual(
      createdCommunity.data!.createCommunity
    );
  });

  it("does not create a duplicate community", async () => {
    await testClient.createCommunity({
      variables: {
        input: { name: "Mock Community" },
      },
    });

    await expect(
      testClient.createCommunity({
        variables: {
          input: { name: "Mock Community" },
        },
      })
    ).rejects.toThrowErrorMatchingInlineSnapshot(`"Duplicate name"`);
  });
});
