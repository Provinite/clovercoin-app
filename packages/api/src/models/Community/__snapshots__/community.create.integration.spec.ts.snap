// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create commuity creates a community 1`] = `
{
  "data": {
    "createCommunity": {
      "__typename": "Community",
      "id": Any<String>,
      "name": "Test Community",
    },
  },
}
`;
